distributed/shuffle/tests/test_shuffle.py

```diff
@@ -2057,6 +2058,28 @@ async def test_handle_null_partitions_p2p_shuffling(c, s, *workers):
     await check_scheduler_cleanup(s)
 
 
+@gen_cluster(client=True)
+async def test_handle_null_partitions_p2p_shuffling_2(c, s, a, b):
```

Comment:
This test fails only fails 2/3 of the time on `main`, but that's good enough for me.